### DISCUSSION TITLE
Use a single repo instead of multiple separated repos

### DISCUSSION THREAD

**Discussion by nmpereira** [CommunityFocus/CommunityFocus#103] [Polls]
## Problem statement
We currently have separated repos (cf-frontend, cf-backend, CommunityFocus). This causes multiple problems that are outlined below. I'm considering merging both (frontend, backend) or all 3 repos. Would like some opinions on if we should do it or not. The problems we currently face, in order of importance:

1. The preview builds on PRs are currently built separately (cf-frontend for a PR uses prod cf-backend, so a feature change that changes both codebases are much harder test). Having a single codebase will make testing out a feature using preview builds much easier)
2. Testing: For UI testing with Cypress, running tests on separated codebases is much harder. If the frontend PR has some code that uses a backend code that isn't merged yet, the tests will fail.
3. Simplified dev experience: Using a single codebase will help with developer exp since only one codebase will need to be looked at instead of multiple.

## Current state
We currently have 3 repos
1. cf-frontend: https://github.com/CommunityFocus/cf-frontend
2. cf-backend: https://github.com/CommunityFocus/cf-backend
3. CommunityFocus: https://github.com/CommunityFocus/CommunityFocus

## Proposed state
1. CommunityFocus/CommunityFocus: will contain all the code, Issues, discussions, etc

## Downsides of the migration
1. Migration will be hard/confusing
2. Work will need to be paused until complete
3. Typescript/ESlint configs might need to be updated, if any

## FAQ
Why did you not use a single repo, to begin with?
1. We thought the experience for new devs was going to be better since they only have to look at 1 repo at a time. 
2. I assumed we needed separated repos to run frontend on netlify and backend on railway, but that's not the case. 
4. Bad planning?: Did not foresee that testing UI would be harder.

**Comment by Caleb-Cohen** [comment]
I prefer a mono-repo and it's probably the best play. That said, working between the separate repos has not been a big issue for me personally.
